tests/fault_tolerance/test_vllm_health_check.py

```
@@ -192,25 +191,53 @@ def test_vllm_health_check_active(request, runtime_services):
-            # Step 5: Send a request triggering the handler to shutdown everything.
-            test_response = send_completion_request("How old are you?", 100, timeout=60)
-            logger.error(f"Test request failed: {test_response}")
-
-            # Step 6: Ensure the worker process has been stopped as a result of the EngineDeadError condition.
-            if worker.is_running():
+            # Wait for the vLLM engine process to actually die
+            if vllm_child:
+                try:
+                    vllm_child.wait(timeout=30)
+                    logger.info(f"vLLM engine process {vllm_child.pid} has been terminated")
+                except Exception as e:
```

Ruff (0.14.3)

Do not catch blind exception: `Exception`

(BLE001)